src/hooks/useEffect.js

```diff
@@ -40,6 +41,11 @@ module.exports = (globalSpace, storeName = TRAM_EFFECT_QUEUE, workingKeyName = T
     // this makes successive useEffects calls unique (until we reset the key)
     incrementWorkingKeyBranch(globalSpace, workingKeyName)
 
-    effectStore[key] = onEffect
+    // if we have triggers, append them to the key
+    // this will make calls with new / different triggers to restart the effect
+    const formatTriggers = triggers.join(':')
+    const keyWithTriggers = `${key}(${formatTriggers})`
+
+    effectStore[keyWithTriggers] = onEffect
```

Comment:
this is the meat of the change.
Basically, appending the triggers to the key, which means that if the triggers change, then on finishing the mount, we'll have noticed that one key no longer exists, and a new key that we've never seen before does, causing the respective cleanup and start of the effects.